data/crac-creation/crac-creator-cse/src/test/java/com/farao_community/farao/data/crac_creation/creator/cse/CseCracCreatorWithMneTest.java

```diff
@@ -0,0 +1,204 @@
+        } else {
+            assertEquals(fromNode + "  " + toNode + "  " + suffix, crac.getFlowCnec(createdCnecId).getNetworkElement().getId());
+
+        }
+        assertEquals(crac.getFlowCnec(createdCnecId).getNominalVoltage(Side.RIGHT), crac.getFlowCnec(createdCnecId).getNominalVoltage(Side.RIGHT));
+        assertEquals(expectedNV, crac.getFlowCnec(createdCnecId).getNominalVoltage(Side.RIGHT), 0.00001);
+    }
+
+    public void assertMneBaseCaseInCrac(String name, String fromNode, String toNode, String suffix, String direction, Instant instant, double expectedIMax, double expectedThreshold, Unit expectedThresholdUnit, double expectedNV) {
+        String createdCnecId = String.format("%s - %s ->%s  - %s", name, fromNode, toNode, instant);
+        String nativeId = String.format("%s - %s  - %s  - %s", name, fromNode, toNode, "basecase");
+        Crac crac = cracCreationContext.getCrac();
+        assertNotNull(crac.getFlowCnec(createdCnecId));
+        assertEquals(name, crac.getFlowCnec(createdCnecId).getName());
+        assertTrue(crac.getFlowCnec(createdCnecId).isMonitored());
+        assertFalse(crac.getFlowCnec(createdCnecId).isOptimized());
+        assertEquals(expectedIMax, crac.getFlowCnec(createdCnecId).getIMax(Side.LEFT), 0.00001);
+        assertEquals(expectedIMax, crac.getFlowCnec(createdCnecId).getIMax(Side.RIGHT), 0.00001);
+        FlowCnec flowCnec = crac.getFlowCnec(createdCnecId);
+        assertTrue(hasThreshold(nativeId, expectedThreshold, expectedThresholdUnit, flowCnec, direction, Side.LEFT));
+        assertTrue(hasThreshold(nativeId, expectedThreshold, expectedThresholdUnit, flowCnec, direction, Side.RIGHT));
+        assertTrue(crac.getFlowCnec(createdCnecId).getState().isPreventive());
+        assertEquals(crac.getFlowCnec(createdCnecId).getNetworkElement().getId(), crac.getFlowCnec(createdCnecId).getNetworkElement().getName());
+        boolean directionInvertedInNetwork = cracCreationContext.getBranchCnecCreationContext(nativeId).isDirectionInvertedInNetwork();
+        if (directionInvertedInNetwork) {
+            assertEquals(toNode + "  " + fromNode + "  " + suffix, crac.getFlowCnec(createdCnecId).getNetworkElement().getId());
+        } else {
+            assertEquals(fromNode + "  " + toNode + "  " + suffix, crac.getFlowCnec(createdCnecId).getNetworkElement().getId());
+        }
+        assertEquals(crac.getFlowCnec(createdCnecId).getNominalVoltage(Side.RIGHT), crac.getFlowCnec(createdCnecId).getNominalVoltage(Side.RIGHT));
+        assertEquals(expectedNV, crac.getFlowCnec(createdCnecId).getNominalVoltage(Side.RIGHT), 0.00001);
+    }
+
+    private boolean hasThreshold(String nativeId, double expectedThreshold, Unit expectedThresholdUnit, FlowCnec flowCnec, String direction, Side side) {
+        boolean directionInvertedInNetwork = cracCreationContext.getBranchCnecCreationContext(nativeId).isDirectionInvertedInNetwork();
+        if (direction.equals("DIRECT")) {
```

Comment:
maybe add a new boolean "direct" depending on the values of directionInvertedInNetwork & direction.
if directionInvertedInNetwork && "DIRECT" OR !directionInvertedInNetwork && "OPPOSITE" => direct = true
if !directionInvertedInNetwork && "DIRECT" OR directionInvertedInNetwork && "OPPOSITE" => direct = false

Then if direct = true, test max only
If direct = false, test min only
This way you can simplify lines 156 & 162.

Of course, you will have to keep the last case as is (lines 166 - 171)